bigbluebutton-html5/imports/api/users/server/methods/userLeaving.js

```diff
@@ -23,38 +21,22 @@ export default function userLeaving(credentials, userId) {
   };
 
   const User = Users.findOne(selector);
+
   if (!User) {
     throw new Meteor.Error('user-not-found', `Could not find ${userId} in ${meetingId}: cannot complete userLeaving`);
   }
 
-  if (User.connectionStatus === OFFLINE_CONNECTION_STATUS) {
-    return null;
-  }
-
-  if (User.validated) {
-    const modifier = {
-      $set: {
-        validated: null,
-      },
-    };
-
-    const cb = (err) => {
-      if (err) {
-        return Logger.error(`Invalidating user: ${err}`);
-      }
-
-      return Logger.info(`Invalidate user=${userId} meeting=${meetingId}`);
-    };
-
-    Users.update(selector, modifier, cb);
+  // If the current user connection is not the same that triggered the leave we skip
+  if (User.connectionId !== connectionId) {
+    return false;
   }
```

Comment:
The deleted code was not actually needed since we clear the validated on `UserLeftMeetingEvtMsg` which is triggered by this publish.

Reply:
Will this work consistently with page refresh?
There might be a race condition here, since Meteor is waiting for the userLeave message from akka, but he client could try to subscribe before that.

Reply:
I can see that happening. I can`t really mimic heavy load to actually make a race condition :/